lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -214,34 +213,29 @@ private ColumnMetaData getColumnMetaData()
     private void flushCurrentPageToBuffer()
             throws IOException
     {
-        ImmutableList.Builder<ParquetDataOutput> outputDataStreams = ImmutableList.builder();
-
-        BytesInput bytesInput = BytesInput.concat(copy(repetitionLevelWriter.getBytes()),
-                copy(definitionLevelWriter.getBytes()),
-                copy(primitiveValueWriter.getBytes()));
-        ParquetDataOutput pageData = (compressor != null) ? compressor.compress(bytesInput) : createDataOutput(bytesInput);
+        BytesInput bytesInput = BytesInput.concat(
+                repetitionLevelWriter.getBytes(),
+                definitionLevelWriter.getBytes(),
+                primitiveValueWriter.getBytes());
         long uncompressedSize = bytesInput.size();
+        ParquetDataOutput pageData = (compressor != null)
+                ? compressor.compress(bytesInput)
+                : createDataOutput(Slices.wrappedBuffer(bytesInput.toByteArray()));
```

Comment:
`toByteArray` will perform copy. Is it needed?

Reply:
The SequenceBytesIn returned by BytesInput.concat was causing test failures unless it's explicitly copied, the compressor was doing it internally but it had to be done explicitly for the uncompressed case. I've now simplified the code to make things more consistent.